shared/modules/selectors/networks.ts

```diff
@@ -43,6 +45,81 @@ export function getSelectedNetworkClientId(
 ) {
   return state.metamask.selectedNetworkClientId;
 }
+/**
+ * Type to extend InternalNetworkState with multichain configurations.
+ */
+type ExtendedInternalNetworkState = InternalNetworkState & {
+  multichainNetworkConfigurationsByChainId: Record<
+    string,
+    NetworkConfiguration
+  >;
+};
+
+/**
+ * State type that includes both standard and multichain network configurations.
+ */
+export type ConsolidatedNetworkConfigurationsState = {
+  metamask: Pick<
+    ExtendedInternalNetworkState,
+    | 'networkConfigurationsByChainId'
+    | 'multichainNetworkConfigurationsByChainId'
+  >;
+};
+
+/**
+ * Combines and returns network configurations for all chains (EVM and not).
+ *
+ * @param state - Redux state.
+ * @returns A consolidated object containing all available network configurations.
+ */
+export const getAllNetworkConfigurationsByCaipChainId = createDeepEqualSelector(
+  (state: ConsolidatedNetworkConfigurationsState) =>
+    state.metamask.networkConfigurationsByChainId,
+  (state: ConsolidatedNetworkConfigurationsState) =>
+    state.metamask.multichainNetworkConfigurationsByChainId,
+  (state: {
+    metamask: { internalAccounts: AccountsControllerState['internalAccounts'] };
+  }) => state.metamask.internalAccounts,
+  (
+    networkConfigurationsByChainId,
+    multichainNetworkConfigurationsByChainId,
+    internalAccounts,
+  ) => {
+    const caipFormattedEvmNetworkConfigurations: Record<
+      string,
+      NetworkConfiguration
+    > = {};
+
+    Object.entries(networkConfigurationsByChainId).forEach(
+      ([chainId, network]) => {
+        const caipChainId = `eip155:${hexToDecimal(chainId)}`;
+        caipFormattedEvmNetworkConfigurations[caipChainId] = network;
+      },
+    );
+
+    Object.entries(multichainNetworkConfigurationsByChainId).forEach(
+      ([caipChainId, networkConfig]) => {
+        const matchesAccount = Object.values(internalAccounts.accounts).some(
+          (account) => {
+            const matchesScope = account.scopes.some((scope) => {
+              return scope === caipChainId;
+            });
+
+            const isSnapEnabled = account.metadata.snap?.enabled;
+
+            return matchesScope && isSnapEnabled;
+          },
+        );
+
+        if (matchesAccount) {
+          caipFormattedEvmNetworkConfigurations[caipChainId] = networkConfig;
+        }
+      },
+    );
```

Comment:
should probably leave a comment by this filtering logic is needed and why it filters this way

Reply:
https://github.com/MetaMask/metamask-extension/pull/31326/files#r2025160850